### ISSUE TITLE
add support signing container images by using cosign project

### ISSUE THREAD

**Issue by developer-guy** [carvel-dev/kbld#177]
**Describe the problem/challenge you have**
I would like to sign container images while building them through carvel-kbld.

**Describe the solution you'd like**
[cosign](https://github.com/sigstore/cosign) maintained by [sigstore](https://sigstore.dev/) community is a new standard that allows the signature and verification of container images and OCI artifacts.

we (w/@dentrax @erkanzileli) thought that we can add this support for signing images right after built them.

**Anything else you would like to add:**
[Additional information that will assist in solving the issue.]

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by cppforlife** [comment]
@developer-guy yes! we are definitely very open to contribution of such feature.

x-link (existing issue): https://github.com/vmware-tanzu/carvel-kbld/issues/157
